src/sentry/runner/commands/createuser.py

```diff
@@ -37,97 +37,124 @@ def _get_superuser():
+        if not email:
+            raise click.ClickException("Invalid or missing email address.")
 
-            # if we've only got a single team let's go ahead and give
-            # access to that team as its likely the desired outcome
-            teams = list(Team.objects.filter(organization=org)[0:2])
-            if len(teams) == 1:
-                OrganizationMemberTeam.objects.create(team=teams[0], organizationmember=member)
-            click.echo(f"Added to organization: {org.slug}")
+        if not no_password and not password:
+            raise click.ClickException("No password set and --no-password not passed.")
 
-    if password:
-        user.set_password(password)
-        user.save()
+        from sentry import roles
+        from sentry.models import User
 
-    # for self hosted to give superusers admin permissions
-    if superuser is True and settings.SENTRY_SELF_HOSTED and not no_input:
-        _set_user_permissions(user)
+        fields = dict(
+            email=email, username=email, is_superuser=superuser, is_staff=staff, is_active=True
+        )
 
-    click.echo(f"User {verb}: {email}")
+        verb = None
+        try:
+            user = User.objects.get(username=email)
+        except User.DoesNotExist:
+            user = None
+
+        # Update the user if they already exist.
+        if user is not None:
+            if force_update:
+                user.update(**fields)
+                verb = "updated"
+            else:
+                click.echo(f"User: {email} exists, use --force-update to force")
+                sys.exit(3)
+        # Create a new user if they don't already exist.
+        else:
+            user = User.objects.create(**fields)
+            verb = "created"
+
+            # TODO(dcramer): kill this when we improve flows
+            if settings.SENTRY_SINGLE_ORGANIZATION:
+                from sentry.models import (
+                    Organization,
+                    OrganizationMember,
+                    OrganizationMemberTeam,
+                    Team,
+                )
+
+                org = Organization.get_default()
+                if superuser:
+                    role = roles.get_top_dog().id
+                else:
+                    role = org.default_role
+                member = OrganizationMember.objects.create(organization=org, user=user, role=role)
+
+                # if we've only got a single team let's go ahead and give
+                # access to that team as its likely the desired outcome
+                teams = list(Team.objects.filter(organization=org)[0:2])
+                if len(teams) == 1:
+                    OrganizationMemberTeam.objects.create(team=teams[0], organizationmember=member)
+                click.echo(f"Added to organization: {org.slug}")
+
+        if password:
+            user.set_password(password)
+            user.save()
+
+        if superuser:
+            _set_superadmin(user)
```

Comment:
I'd like to see a comment here or in the `_set_superadmin` function documenting that superadmin is a kludge that approximates `superuser` using the role system vs. a model attribute.